### PR TITLE
Update organization_header and create tests

### PR DESCRIPTION
**Pull request by marlo-longley**
~~This is pretty much ready to go, but could use some input on the contact_email test if anyone has a minute, thanks! I commented out that `it` block in the test file.~~

### CODE REVIEW

**Comment by marlo-longley** [comment]
Fixes #659

File: app/views/shared/_organization_header.html.erb [hunk 1]
```diff
@@ -13,14 +13,18 @@
...
         <% end %>
         <% if @organization.contact_emails.any? %>
-        <span class="lead"><%= mail_to @organization.contact_emails.first, t('.contact') %></span>
+        <span class="lead"><%= mail_to @organization.contact_emails.first.email, t('.contact') %></span>
         <% end %>
       </div>
       <div class="ms-auto">
-        <% if current_page?(organization_path(@organization)) %>
-        <%= link_to can?(:edit, @organization) ? t('.manage_org') : t('.view_org'), organization_users_path(@organization), class: 'align-self-center' %>
+        <% if current_page?(organization_path(@organization)) ||
```
Comment (corylown) [inline app/views/shared/_organization_header.html.erb:20]: I think with some partial re-organization you could simplify the logic here with a `yield`

`<%= content_for :org_header_section_link %>`

**Review by corylown — CHANGES_REQUESTED** [review]
@marlo-longley the logic in the org header to decide which link to show is suggesting to me that a re-organization of partials might provide a better solution. There are really two layouts needed -- one for the provider page set of pages and another for the manage organization set of pages.

- Create two partials. One for the provider page and one for the manage set of pages. The provider page one might look like this. Note the content_for block at the top that sets the appropriate org header link for the section.:
```
# views/organizations/_provider_page.html.erb
<% content_for :org_header_section_link do %>
  <div class="ms-auto">
    <% if @organization.provider? %>
      <%= link_to t('.default_stream'), organization_path(@organization), class: 'align-self-center' %>
    <% end %>
  </div>
<% end %>

<%= render 'shared/organization_header' %>

<div class="container manage-organization-container">
  <%= render 'streams/header', stream: @organization.default_stream %>
  <%= render 'shared/provider_page_header' %>
  <%= yield %>
</div>

```

- Change the separate partials for each page of the provider and manage org pages so that they call the appropriate generic partial for their section. For the normalized_data page it might look like:

```
# views/organizations/normalized_data.html.erb
<%= render 'provider_page'  do %>
  <h3>Normalized data</h3>
   <%= render 'streams/normalized_dump', normalized_dump: @organization.default_stream.normalized_dumps.full_dumps.last %>
<% end %>
```

- replace the linking logic in the header with a content_for:
`<%= content_for :org_header_section_link %>`

**Comment by corylown** [comment]
As a sort of general rule if I find myself wanting to use `current_page` logic in a template that's a hint to refactor.

File: app/views/organizations/processing_status.html.erb [hunk 2]
```diff
@@ -1,11 +1,4 @@
-<%= render 'shared/organization_header' %>
-
-<div class="container manage-organization-container">
-  <%= render 'streams/header', stream: @organization.default_stream %>
-  <%= render 'shared/provider_page_header'%>
-
-  <h3><%= t('job_tracker.title') %></h3>
-
+<%= render 'shared/layout_provider_page' do %>
+  <<h3><%= t('job_tracker.title') %></h3>
```
Comment (corylown) [inline app/views/organizations/processing_status.html.erb:2]: extra `<`:
```suggestion
  <h3><%= t('job_tracker.title') %></h3>
```

File: app/views/organization_users/index.html.erb [hunk 3]
```diff
@@ -1,6 +1,4 @@
-<%= render 'shared/organization_header' %>
-<div class="container manage-organization-container">
-    <%= render 'shared/manage_organization_header' %>
+<%= render 'shared/layout_manage_organization' do %>
     <h3><%= t('.title') %></h3>
```
Comment (corylown) [inline app/views/organization_users/index.html.erb:2]: I think you're going to need to fully qualify the translation paths now.

```suggestion
    <h3><%= t('organization.index.title') %></h3>
```

**Review by corylown — CHANGES_REQUESTED** [review]
Overall, looks good. Check the translation paths in the templates you updated. I think you'll have to fully qualify the ones in a render block. I pointed out one instance, but there are others I think.

**Review by corylown — APPROVED** [review]

**Pull request merged**